recipes/libcurl/all/conanfile.py

```diff
@@ -296,14 +296,43 @@ def _patch_autotools(self):
     def _patch_cmake(self):
         if not self._is_using_cmake_build:
             return
+        cmakelists = os.path.join(self._source_subfolder, "CMakeLists.txt")
         # Custom findZstd.cmake file relies on pkg-config file, make sure that it's consumed on all platforms
         if self._has_zstd_option:
             tools.replace_in_file(os.path.join(self._source_subfolder, "CMake", "FindZstd.cmake"),
                                   "if(UNIX)", "if(TRUE)")
         # TODO: check this patch, it's suspicious
-        tools.replace_in_file(os.path.join(self._source_subfolder, "CMakeLists.txt"),
+        tools.replace_in_file(cmakelists,
                               "include(CurlSymbolHiding)", "")
 
+        # libnghttp2
+        tools.replace_in_file(
+            cmakelists,
+            "find_package(NGHTTP2 REQUIRED)",
+            "find_package(libnghttp2 REQUIRED)",
+        )
+        tools.replace_in_file(
+            cmakelists,
+            "include_directories(${NGHTTP2_INCLUDE_DIRS})",
+            "",
+        )
+        tools.replace_in_file(
+            cmakelists,
+            "list(APPEND CURL_LIBS ${NGHTTP2_LIBRARIES})",
+            "list(APPEND CURL_LIBS libnghttp2::nghttp2)",
+        )
+        if tools.Version(self.version) >= "7.80.0":
+            tools.replace_in_file(
+                cmakelists,
+                'get_target_property(_lib "${_libname}" LOCATION)',
+                """get_target_property(_type "${_libname}" TYPE)
+    if(${_type} STREQUAL "INTERFACE_LIBRARY")
+      # Reading the INTERFACE_LIBRARY property on non-imported target will error out.
+      continue()
+    endif()
+    get_target_property(_lib "${_libname}" LOCATION)""",
+            )
```

Comment:
So this specific part allows to use cmake_find_package? Maybe add a comment because it's not specific to libnghttp2?

Reply:
@SpaceIm Yes, this part allow to use INTERFACE_LIBRARY target, such as generate by the cmake_find_package generator.